AUTOMATAFORMAT.md

```diff
@@ -0,0 +1,111 @@
+
+# The file format for automata
+* Based partially on Ondrej Lengal's [`.vtf' format](https://discord.com/channels/@me/864885374375821312/980792642927460372), a format of Pavol Vargovcik, and a confused disucssion.
+
+## Top-level file structure
+* The format is **line**-based. Lines can be connected by `/`.
+* Lines are parsed into tokens. Tokens are delimited by white spaces, the format is **white space sensitive**.
+* A file contains **sections** starting with a line of the form `@<SECTION-TYPE>`, containing an automaton or, in theory, anything. We will start with one section per file, containing an automaton, but it is obviously extensible to more sections and more kinds of things.
+* In automata sections, non-empty lines are **key-value lines** of the form `%<KEY> [VALUE]`, or **transition lines**. Several key-value lines with the same key mean that the key is mapped to the set of values, an occurence without a value marks the `KEY` as defined. 
+* Besides white spaces and the end of line, the following are **reserved characters**: `&`,`|`,`!`,`@`,`(`,`)`,`%`,`"`,`/`,`\` `[`,`]`,`a`,`q`,`n`,`f`,`{`,`}`.
+  * `&`,`|`,`!`,`(`,`)` occurring as tokens in transition lines are logical operators.
+  * `[`,`]`,`-` within intervals of symbols of the form `[bla-bli]`.
+  * `{`,`}` are enclosing attributes, if we use them this way. Remains to be decided. (?)
+  * `@` opens the line with a section name and is used in transducer alphabet tokens of the form `x@a1`, `y@[10-55]`,...
+  * `%` opens a key-value line, possibly also key-value appendix of a transition. Remains to be decided. (?)
+  * `"` strings containing white spaces and reserved characters can be written in between `"`. The reserved characters lose their special meaning. The characters `"` and `\` inside a string must be escaped, i.e. `\"` and `\\`.
+  * `/` to concatenate lines.
+  * `\` for escaping characters.
+  * `a`,`q`,`n`,`f` are token type specifiers (alphabet, state, node, formula). If you don't like these, propose other ones.
```

Comment:
what's the difference between `n` and `f`?

Reply:
f is a formula that does not adhere to our predefined format of formulas, the intention was to be able to write SMT-lib formulas to specify classes of symbols and relations between transducer tracks. I can perhaps be other things too. The format does not specify what the formula is, it is sort of up to the user.

Reply:
we will use `a` for formulae